src/DataTable/index.jsx

```diff
@@ -1,76 +1,93 @@
-import React from 'react'
-
-import Pagination from './Pagination'
-import Row from './Row'
-import Search from './Search'
+import React, { Fragment } from "react";
+import { getTotalNumberOfPages, search } from "./logic";
+import PaginationBox from "./PaginationBox";
+import Table from "./Table";
+import SearchBox from "./SearchBox";
 
 class DataTable extends React.Component {
   state = {
-    rows: this.props.rows,
-    currentPageNumber: 0,
-    totalNumberOfPages: this.calculateTotalNumberOfPages(this.props.rows)
```

Comment:
never call functions inside state declaration, bad bad practice!!! 
1. you don't know what is the default value this function returns.
2. what if there is an error while calling this function